packages/create-schemas/src/config.ts

```diff
@@ -0,0 +1,81 @@
+import { Command } from "commander";
+import { loadConfig } from "c12";
+import * as z from "zod";
+import packageJson from "../package.json" with { type: "json" };
+import type { OpenAPITSOptions as OriginalOpenAPITSOptions } from "openapi-typescript";
+
+const CONFIG_FILE_DEFAULT = "create-schemas.config";
+const OUTPUT_FILE_DEFAULT = "openapi-types.ts";
+const ROOT_DEFAULT = ".";
+
+type OpenApiTsOptions = Omit<OriginalOpenAPITSOptions, "cwd" | "transform" | "postTransform" | "silent" | "version">;
+
+export interface UserConfig {
+    root?: string;
+    input?: string;
+    output?: string;
+    openApiTsOptions?: OpenApiTsOptions;
+}
+
+export interface InlineConfig extends UserConfig {
+    configFile?: string;
+}
+
+const resolvedConfigSchema = z.object({
+    configFile: z.string(),
+    root: z.string(),
+    input: z.string(),
+    output: z.string(),
+    openApiTsOptions: z.custom<OpenApiTsOptions>().optional().default({})
+});
+
+export type ResolvedConfig = z.infer<typeof resolvedConfigSchema>;
+
+export function parseArgs(argv?: string[]): InlineConfig {
+    const program = new Command();
+
+    program
+        .name("create-schemas")
+        .version(packageJson.version, "-v, --version", "display version number")
+        .argument("[input]")
+        .option("-c, --config <file>", "use specified config file", CONFIG_FILE_DEFAULT)
+        .option("-i, --input <path>", "path to the OpenAPI schema file")
+        .option("-o, --output <path>", "output file path", OUTPUT_FILE_DEFAULT)
+        .option("--cwd <path>", "path to working directory", ROOT_DEFAULT)
```

Comment:
Just realized I shouldn't provide default values here, because I want the values passed to the CLI to take precedence over the values from the config file. If we provide default values, we can't distinguish between values explicitely passed by the user (which should take precedence) and the default values (which should not).